File: frontend-v2/lib/types/task.ts
export interface Task {
  id: string
  title: string
  description: string | null
  status: string
  order: number
  dueDate: string | null
  priority: string
  createdAt: string
  updatedAt: string
  userId: string
  assigneeId: string | null
  parentId: string | null
  subtasks?: Task[]
  tags?: Tag[]
  user?: User
  assignee?: User | null
  parent?: Task | null
}

export interface Tag {
  id: string
  name: string
  color: string | null
  createdAt: string
  updatedAt: string
}

export interface User {
  id: string
  email: string
  firstName: string | null
  lastName: string | null
  createdAt: string
  updatedAt: string
}

File: DevTask/app/src/main/java/dev/espera/devtask/data/datasources/remote/models/Task.kt
package dev.espera.devtask.data.datasources.remote.models

import dev.espera.devtask.data.common.TaskPriority
import dev.espera.devtask.data.common.TaskStatus

data class Task(
    val assignee: User?,
    val assigneeId: String?,
    val createdAt: String,
    val description: String,
    val dueDate: String,
    val id: String,
    val order: Int,
    val parent: Task?,
    val parentId: String?,
    val priority: TaskPriority,
    val status: TaskStatus,
    val subtasks: List<Task> = emptyList(),
    val tags: List<Tag> = emptyList(),
    val title: String,
    val updatedAt: String,
    val userId: String
)